b1ddi/resource_dns_auth_zone.go

```diff
@@ -624,3 +630,36 @@ func resourceConfigAuthZoneDelete(ctx context.Context, d *schema.ResourceData, m
 	d.SetId("")
 	return nil
 }
+
+func inheritanceSourceObjUpdater(d []interface{}, r *models.ConfigAuthZoneInheritance) *models.ConfigAuthZoneInheritance {
+	if d == nil || len(d) == 0 || r == nil {
+		return nil
+	}
+	authZoneInheritance := new(models.ConfigAuthZoneInheritance)
+
+	for _, v := range d {
+		for key, value := range v.(map[string]interface{}) {
```

Comment:
It might be safer to check that the `v.(map[string]interface{})` cast works before using the returned value.

Reply:
Done